node/src/main/scala/coop/rchain/node/configuration/model.scala

```diff
@@ -44,12 +44,14 @@ final case class ProtocolServer(
     port: Int,
     grpcMaxRecvMessageSize: Long,
     grpcMaxRecvStreamMessageSize: Long,
-    maxMessageConsumers: Int
+    maxMessageConsumers: Int,
+    enableStateExporter: Boolean
 )
 
 final case class ProtocolClient(
     networkId: String,
     bootstrap: PeerNode,
+    trimState: Boolean,
```

Comment:
I'm not sure if these new options are directly _protocol_ related. They are more general, maybe on the RSpace level but even more, as the node parameters (*) to determine how state will be shaped and accessed. Transport layer or RSpace will not change its behavior depending on these options.

\* Because we have Engine, where most components are connected, inside casper project, maybe it's easiest solution is to be part of casper options.

Reply:
@tgrospic I tend not to agree here - this config and CLI options are user facing features. So looking from node operator standpoint, `trimState` is a parameter that accompanying `bootstrap`, and saying that node operator does not want to download the whole state (or he does in case of `false`).  We should make configuration as simple as possible without assuming node operator has some implicit knowledge about the platform. We have protocol server and API server in node software, everything that talks with protocol server of another node - is protocol client related here, IMO.

Reply:
Good point. This section is more _client_ then _protocol_ so it makes sense. Maybe `requestTrimState` would be appropriate?

Reply:
`requestTrimState ` sounds good to me, but imo no too different from `trimState`. I don't se something else that can be called `trimState` in this context